### PR TITLE
Add workflow to run the tests on arm64 macos and windows

### PR DESCRIPTION
**Pull request by cr-xu**
## Description

Add a Github workflow to run pytest on arm64 macos as requested in #223. 

However, the Github runner doesn't support MPS yet, so it cannot be tested. c.f. https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#limitations-for-arm64-macos-runners
> Nested-virtualization and Metal Performance Shaders (MPS) are not supported due to the limitation of Apple's Virtualization Framework.

## Motivation and Context

- [x] I have raised an issue to propose this change (required for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [ ] I have updated the documentation accordingly.
  
  
- [ ] I have reformatted the code and checked that formatting passes (**required**).
- [x] I have have fixed all issues found by `flake8` (**required**).
- [x] I have ensured that all `pytest` tests pass (**required**).
- [ ] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [x] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line

### CODE REVIEW

**Comment by cr-xu** [comment]
An annoying fact is that even though MPS is not really supported in the GitHub runner, it still can be detected by `torch.backends.mps.is_available()`, which causes the test to fail.

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import torch
+
+
+# Check if MPS is available and functional (for GitHub Actions)
+def is_mps_available_and_functional():
+    if not torch.backends.mps.is_available():
```
Comment (jank324) [inline tests/__init__.py:6]:
```suggestion
def is_mps_available_and_functional():
    """Check if MPS is available and functional (for GitHub Actions)."""
    if not torch.backends.mps.is_available():
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +38,7 @@
 - Added imports to the code example in the README (see #188) (@jank324)
 - Refactor definitions of physical constants (see #189) (@hespe)
 - Fix the quadrupole strength units in the quadrupole docstring (see #202) (@ansantam)
+- Allow CI to run also on arm64 macOS and Windows (see #226) (@cr-xu)
```
Comment (jank324) [inline CHANGELOG.md:41]:
```suggestion
- Allow CI to run also on macOS (arm64) and Windows (see #226) (@cr-xu)
```
Just to be clear that it doesn't mean Windows on ARM

File: .github/workflows/macos_test.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: macos-pytest
```
Comment (jank324) [inline .github/workflows/macos_test.yaml:1]:
```suggestion
name: pytest-macos
```

Comment (jank324) [reply .github/workflows/macos_test.yaml:1]: For better sorting

File: .github/workflows/windows_test.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: windows-pytest
```
Comment (jank324) [inline .github/workflows/windows_test.yaml:1]:
```suggestion
name: pytest-windows
```

File: tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import torch
+
+
+# Check if MPS is available and functional (for GitHub Actions)
+def is_mps_available_and_functional():
```
Comment (jank324) [inline tests/__init__.py:5]: Should this be in `tests.__init__.py` or maybe in `tests.utils.py` or even `cheetah.utils`?

Comment (cr-xu) [reply tests/__init__.py]: I moved it to `cheetah.utils` now

File: CHANGELOG.md [hunk 6]
```diff
@@ -38,6 +38,7 @@
 - Added imports to the code example in the README (see #188) (@jank324)
 - Refactor definitions of physical constants (see #189) (@hespe)
 - Fix the quadrupole strength units in the quadrupole docstring (see #202) (@ansantam)
+- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu)
```
Comment (jank324) [inline CHANGELOG.md:41]:
```suggestion
- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu, @jank324, @Hespe)
```

**Review by jank324 — APPROVED** [review]

**Pull request merged**